src/assert.js

```diff
@@ -340,9 +340,17 @@ class Assert {
 				expected = errorString( expected );
 
 			// Expected is a validation function which returns true if validation passed
-			} else if ( expectedType === "function" && expected.call( {}, actual ) === true ) {
-				expected = null;
-				result = true;
+			} else if ( expectedType === "function" ) {
+
+				// protect against accidental semantics which could hard error in the test
+				try {
+					result = expected.call( {}, actual ) === true;
```

Comment:
result needs to be a boolean I think, so may need to assin in the `catch` es well as otherwise I think it would remain `undefined`.

Reply:
The `result` var is actually initialized to `false`, so the "nothing matched" path is a big fallthrough to just default to false.

Reply:
Aye, thanks. I missed that.